plerkle_snapshot/src/bin/solana-snapshot-etl/geyser.rs

```diff
@@ -40,46 +34,100 @@ impl GeyserDumper {
             .with_style(spinner_style)
             .with_prefix("accs");
 
+        let mut connection_config = Map::new();
+        connection_config.insert(
+            "redis_connection_str".to_owned(),
+            figment::value::Value::String(
+                Tag::default(),
+                std::env::var("SNAPSHOT_REDIS_CONNECTION_STR").expect(
+                    "SNAPSHOT_REDIS_CONNECTION_STR must be present for snapshot processing",
+                ),
+            ),
+        );
+        let mut messenger = RedisMessenger::new(MessengerConfig {
+            messenger_type: plerkle_messenger::MessengerType::Redis,
+            connection_config,
+        })
+        .await
+        .expect("create redis messenger");
+        messenger
+            .add_stream(ACCOUNT_STREAM_KEY)
+            .await
+            .expect("configure accounts stream");
+        messenger
+            .set_buffer_size(ACCOUNT_STREAM_KEY, 100_000_000)
+            .await;
+
         Self {
+            messenger: Arc::new(Mutex::new(messenger)),
             accounts_spinner,
-            plugin,
-            accounts_count: 0,
+            accounts_count: Arc::new(AtomicU64::new(0)),
             throttle_nanos,
         }
     }
 
-    pub(crate) fn dump_account(
+    pub async fn dump_account(
         &mut self,
-        account: StoredAccountMeta,
+        (meta, account): (StoredMeta, AccountSharedData),
         slot: u64,
     ) -> Result<(), Box<dyn Error>> {
-        self.plugin.update_account(
-            ReplicaAccountInfoVersions::V0_0_1(&ReplicaAccountInfo {
-                pubkey: account.meta.pubkey.as_ref(),
-                lamports: account.account_meta.lamports,
-                owner: account.account_meta.owner.as_ref(),
-                executable: account.account_meta.executable,
-                rent_epoch: account.account_meta.rent_epoch,
-                data: account.data,
-                write_version: account.meta.write_version,
-            }),
-            slot,
-            /* is_startup */ false,
-        )?;
-        self.accounts_count += 1;
-        if self.accounts_count % 1024 == 0 {
-            self.accounts_spinner.set_position(self.accounts_count);
-        }
+        let account: Account = account.into();
+        // Get runtime and sender channel.
+        // Serialize data.
+        let ai = &ReplicaAccountInfo {
+            pubkey: meta.pubkey.as_ref(),
+            lamports: account.lamports,
+            owner: account.owner.as_ref(),
+            executable: account.executable,
+            rent_epoch: account.rent_epoch,
+            data: &account.data,
+            write_version: meta.write_version,
+        };
+        let account =
+            plerkle_serialization::solana_geyser_plugin_interface_shims::ReplicaAccountInfoV2 {
```

Comment:
Can we move this type into `use`?

Reply:
copied straight from the plugin, don't think it's worth it tbh.